src/cargo/core/registry.rs

```diff
@@ -476,12 +477,21 @@ impl<'gctx> PackageRegistry<'gctx> {
             let name = summary.package_id().name();
             let version = summary.package_id().version();
             if !name_and_version.insert((name, version)) {
-                bail!(
-                    "cannot have two `[patch]` entries which both resolve \
-                     to `{} v{}`",
+                let duplicate_locations: Vec<_> = patch_deps
+                    .iter()
+                    .filter(|&p| p.0.dep.package_name() == name)
+                    .map(|p| p.0.loc.to_string())
+                    .unique()
+                    .collect();
+                return Err(anyhow::anyhow!(
+                    "Cannot have two `[patch]` entries which both resolve to `{} v{}`.\n\
+                    Check patch definitions for `{}` in `{}`",
                     name,
-                    version
-                );
+                    version,
+                    name,
+                    duplicate_locations.join(", ")
+                ))
+                .context(format!("failed to resolve patches in `{}`", url));
```

Comment:
Unless there is a driving reason to do so, I think it would be better to not add this additional context through `context` as it makes things less flexible for how we could improve the wording of the error messages, see also https://github.com/rust-lang/cargo/pull/16407/commits/a39d5c0afc76ae14d1cd2d2884605057c39f1611#r2628751019

Reply:
I was thinking about unifying the error context for similar patch resolve problems. There are two similar messages with the same additional context in the very same function on lines [L447](https://github.com/rust-lang/cargo/blob/47761c18493f8673626a82de5619c90ace39e550/src/cargo/core/registry.rs#L447) and [L466](https://github.com/rust-lang/cargo/blob/47761c18493f8673626a82de5619c90ace39e550/src/cargo/core/registry.rs#L466)

I fixed preposition `in -> for` in the additional commit to match those two exactly. Although, I see the reasoning. Will make sense to remove this extra context from here and from two other places mentioned above as well?